module-2/backend/week_one.md

```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+  Consistency throughout the code.  Makes for easier debugging/troubleshooting.
+  
 6. What types of variables are accessible in our view templates without explicitly passing them?
+  Instance, Local, and Class variables.
+
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+    @count = 1
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
+    ```ruby
+  get '/horses' do
+    name = 'Mr. Ed'
+    @count = 1
+    erb :index
+  end
+  ```
+
 9. What's the purpose of ERB?
+  It is utilized by sinatra to allow Ruby code to be passed through CSS & HTML.
+  
 10. Why do I need a development AND test database?
+  Such that you do not contaminate the test assertions by making permanant alterations to the database.
+
 11. What's responsive design?
+  When software looks for the client's display requirements and changes the display accordingly to the user's display needs.
+  
 12. What is CRUD and why is it important?
+  It gives us a framework to build our main controller file.
+
 13. What does HTTP stand for? 
+  Hypertext Transfer Protocol
+  
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+  Through the use of the <%= %> tag
+  
 15. What's an ORM?
+  Object Relation Mapping
 16. What's the most commonly used ORM?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+  C++  ??
+
+17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full     CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+  
+  ##See all restaurants.  It is a Read process.##
+  get '/restaurants' do
+    @restraunts = Restraunt.all
+    erb :index.erb
+  end
+  
+  ##See one restaurant.  It is a Read Process##
+  get '/restaurants/:id' do |id|
+    @restaurants = Restaurant.find(id)
+    erb :show.erb
+  end
+  
+  ##View Form to enter new Restaurant.  It is a Create Process.##
+  get '/restaurants/new' do
+    erb :new.erb
+  end
+  
+  ##Submit and save new Restaurant.  It is a Create Process.##
+  post '/restaurants' do
+    Restaurant.create(params[:restaurant])
+    redirect '/restaurants'
+  end
+  
+  ##See Form to edit already existing Restaurant.  It is an Update Process.##
+  get '/restaurants/:id/edit' do |id|
+    @restaurants = Restaurant.find(id)
+    erb :edit.erb
+  end
+  
+  ##Click Submit to update a Restaurant.  It is an Update Process.##
+  put '/restaurants/:id' do |id|
+    Restaurant.update(params[:restaurant], id)
+    redirect '/restaurants/:id'

```

Comment:
we'll want to interpolate the `:id` here or else it will literally go to "/restaurants/:id" which isn't what we want. we want it to go to "/restaurants/5" or whatever restaurant we're talking about so we can do something like: `redirect "/restaurants/#{id}"`